ibis/backends/mysql/converter.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+import datetime
+
+from ibis.formats.pandas import PandasData
+
+
+class MySQLPandasData(PandasData):
+    @classmethod
+    def convert_Time(cls, s, dtype, pandas_type):
+        def convert(timedelta):
+            comps = timedelta.components
+            return datetime.time(
+                hour=comps.hours,
+                minute=comps.minutes,
+                second=comps.seconds,
+                microsecond=comps.microseconds,
+            )
+
+        return s.map(convert, na_action="ignore")
+
+    @classmethod
+    def convert_Timestamp(cls, s, dtype, pandas_type):
+        if s.dtype == "object":
+            s = s.replace("0000-00-00 00:00:00", None)
```

Comment:
This existed before but as a `result_processor` method on a custom sqlalchemy type. This is a good example of something that goes through many fewer layers of software by removing sqlalchemy from the mix here.